BitwardenShared/Core/Platform/Services/Stores/AppSettingsStore.swift

```diff
@@ -131,6 +131,18 @@ protocol AppSettingsStore: AnyObject {
     ///
     func encryptedUserKey(userId: String) -> String?
 
+    /// Retrieves a feature flag value from the app's settings store.
+    ///
+    /// This method fetches the value for a specified feature flag from the app's settings store.
+    /// The value is decoded into a specified type that conforms to `Codable`. If the flag does not exist
+    /// or cannot be decoded, the method returns `nil`.
+    ///
+    /// - Parameter name: The name of the feature flag to retrieve, represented as a `String`.
+    /// - Returns: The value of the feature flag as the specified type `T`, or `nil` if the flag does not exist
+    ///     or cannot be decoded.
+    ///
+    func featureFlag(name: String) -> Bool?
```

Comment:
⛏️ The docs indicate that it's a generic type T, but all we have here is Bool. This also means we don't have overrides for Int or String flags.

Reply:
good catch, yeah for now I kept them as booleans but I can add capability for other types.  I just didn't want to hold this up for testing the onboarding flow but I can certainly tackle that.  Would another ticket be okay with you?

Reply:
Future tickets for the additional types is reasonable to keep this a bit smaller and focused.